src/indev/lv_indev.c

```diff
@@ -1362,6 +1365,35 @@ static void indev_proc_release(lv_indev_t * indev)
     }
 }
 
+static lv_result_t indev_proc_short_click(lv_indev_t * indev)
+{
+    /*Simple short click*/
+    lv_result_t res = send_event(LV_EVENT_SHORT_CLICKED, indev_act);
+    if(res == LV_RESULT_INVALID) {
+        return res;
+    }
+
+    /*Update streak for clicks within small distance and short time*/
+    int32_t dx = indev->pointer.last_short_click_point.x - indev->pointer.act_point.x;
+    int32_t dy = indev->pointer.last_short_click_point.y - indev->pointer.act_point.y;
+    if(dx * dx + dy * dy > indev->scroll_limit * indev->scroll_limit ||
+       lv_tick_elaps(indev->pointer.last_short_click_timestamp) > indev->long_press_time) {
```

Comment:
Check time is cheaper than distance, it can optimize the processing time for usual short clicks .

```suggestion
    if(lv_tick_elaps(indev->pointer.last_short_click_timestamp) > indev->long_press_time ||
       dx * dx + dy * dy > indev->scroll_limit * indev->scroll_limit
       ) {
```